### PR TITLE
feat(permissions): enrich findWithPagination with condition details for plugins

### PR DESCRIPTION
**Pull request by harryburger**
## 📝 Summary

This pull request enhances the DAO permission model by enriching the returned permission data with detailed condition information, improving query performance, and expanding test coverage. The main changes include adding aggregation stages to join and process related plugin, setting, and selector permission data, introducing new types for structured permission responses, and updating indexes for efficiency.

**Permission enrichment and aggregation:**
- Enhanced the `DaoPermission.findWithPagination` aggregation pipeline to join related collections (`Plugin`, `SelectorPermission`, `Setting`) and compute a new `condition` field describing the type and parameters of each permission condition (e.g., voting power, membership, execute-selector, or unknown). This logic uses `$lookup`, `$let`, and `$switch` to determine the condition type and relevant data.
- Added new TypeScript interfaces (`IPermissionConditionData`, `IPermissionResponse`) to define the enriched structure of permission responses.

**Indexing improvements:**
- Added compound indexes to `SelectorPermission` and `Setting` models to optimize lookups by `daoAddress`, `network`, `conditionAddress`, and status fields, supporting the new aggregation queries. [[1]](diffhunk://#diff-4cd129b52bf2de4ef14e8cd686d2295419ce2995c59fd5e913db9c55ba525440R68) [[2]](diffhunk://#diff-91cb0620be211d600e6029061dec013f6d1282fca10d1dca75610531cd395236R230)

**Type and import updates:**
- Updated imports in `daoPermission.ts` and test files to include new enums and types needed for the enriched permission logic. [[1]](diffhunk://#diff-16e46f75a8f410e2856840e0964dc3aac40a7ee8e3ca00fbe0c2748ccb63dc9eR11-R13) [[2]](diffhunk://#diff-b34249299f0006ef2e8ae22106e1b2123b19ca551cc59da7011e10840be84d82L3-R3)

**Test coverage:**
- Added comprehensive unit tests for the new permission condition enrichment logic, covering all condition types and edge cases (voting power, membership, execute-selector, unknown, and no condition).

These changes collectively provide richer, more informative permission data for consumers, while ensuring efficiency and correctness in querying and processing DAO permissions.

### 🔍 Code Review
- [ ] Self-reviewed all code changes
- [ ] Ask AI/Copilot code review
- [ ] Removed all debug code, console.logs, and dead code
- [ ] Implemented error handling with try/catch blocks where needed

### 🧪 Testing
- [ ] All test suites pass locally
- [ ] Added comprehensive unit tests for new features
- [ ] Included integration tests for end-to-end scenarios
- [ ] Successfully tested on remote server

### 🔒 Security
- [ ] Verified no secrets or credentials are exposed
- [ ] Reviewed for common security vulnerabilities
- [ ] **Verified commit authorship** - Reviewed all commits to ensure they're from team members (check for compromised accounts)

### CODE REVIEW

File: src/models/schema/daoPermission.ts [hunk 1]
```diff
@@ -212,6 +218,118 @@ export default class DaoPermission extends Model {
...
+                    {
+                      case: { $or: [{ $eq: ['$$pp.matchedProposal', false] }, '$$hasSelectorRows'] },
+                      then: {
+                        conditionType: 'execute-selector',
+                        selectors: '$selectorRows.selector',
+                        targets: '$selectorRows.target',
+                      },
+                    },
+                  ],
+                  default: { conditionType: 'unknown' },
```
Comment (thekidnamedkd) [inline src/models/schema/daoPermission.ts:335]: Nice switch. One case that might bite us: only `tokenVoting` and `multisig` have branches, so any *other* proposal-creating plugin (`lockToVote`, `spp`, `gauge`, `admin`) whose `proposalCreationConditionAddress` matched (`matchedProposal=true`) skips both typed branches — and it can't reach `execute-selector` either, since that needs `matchedProposal=false` or selector rows. So it falls through to `unknown` even though there's a real condition behind it.

Could we either add branches for the other interface types, or at least drop a `// TODO: non-voting/multisig interface types resolve to 'unknown'` so it's a known gap rather than a silent one? The FE branches on these strings, so an `unknown` where they expected `voting-power` just renders blank. WDYT?

File: src/models/schema/daoPermission.ts [hunk 2]
```diff
@@ -212,6 +218,118 @@ export default class DaoPermission extends Model {
...
+                      case: {
+                        $and: [
+                          { $eq: ['$$pp.matchedProposal', true] },
+                          { $eq: ['$$pp.interfaceType', IPluginInterfaceType.multisig] },
+                        ],
+                      },
+                      then: { conditionType: 'membership', onlyListed: '$$ps.onlyListed' },
+                    },
+                    {
+                      case: { $or: [{ $eq: ['$$pp.matchedProposal', false] }, '$$hasSelectorRows'] },
```
Comment (thekidnamedkd) [inline src/models/schema/daoPermission.ts]: Small edge case: this `$or` fires on `matchedProposal=false` even when there are zero allowed selector rows, so we can emit `{ conditionType: 'execute-selector', selectors: [], targets: [] }`. A consumer then renders an execute-selector permission with an empty allowlist, which reads as "allow nothing" rather than "no/unindexed condition".

Maybe we should require rows before claiming execute-selector, and let the empty case fall through to `unknown`?

```suggestion
                      case: '$$hasSelectorRows',
```

(If we did intend the matched-plugin path to count too, `$and: [{ $eq: ['$$pp.matchedProposal', false] }, '$$hasSelectorRows']` keeps that but still drops the empty arrays.)

File: src/models/schema/daoPermission.ts [hunk 3]
```diff
@@ -212,6 +218,118 @@ export default class DaoPermission extends Model {
...
+      {
+        $lookup: {
+          from: ICollectionNames.Setting,
+          let: { plugin: { $first: '$conditionPlugin' } },
+          pipeline: [
+            { $match: { daoAddress: filter.daoAddress, network: filter.network, status: ISettingStatus.active } },
+            {
+              $match: {
+                $expr: {
+                  $and: [{ $eq: ['$$plugin.matchedProposal', true] }, { $eq: ['$pluginAddress', '$$plugin.address'] }],
```
Comment (thekidnamedkd) [inline src/models/schema/daoPermission.ts]: Tiny consistency thing: every other join in this pipeline lowercases both sides (`$toLower`) to be casing-proof, but this one compares `$pluginAddress` to `$$plugin.address` directly. In practice both are checksummed via ethers so it works today — but if either collection ever stores a differently-cased address, `proposalSetting` silently comes back empty and `minVotingPower`/`onlyListed` drop to `undefined`. Maybe match the rest of the pipeline?

```suggestion
                  $and: [{ $eq: ['$$plugin.matchedProposal', true] }, { $eq: [{ $toLower: '$pluginAddress' }, { $toLower: '$$plugin.address' }] }],
```

File: src/models/schema/daoPermission.ts [hunk 4]
```diff
@@ -212,6 +218,118 @@ export default class DaoPermission extends Model {
...
+          as: 'proposalSetting',
+        },
+      },
+      {
+        $addFields: {
+          condition: {
+            $let: {
+              vars: {
+                pp: { $first: '$conditionPlugin' },
+                ps: { $first: '$proposalSetting' },
```
Comment (thekidnamedkd) [inline src/models/schema/daoPermission.ts:293]: Both `pp` (L285) and `ps` (here) take `$first` of a lookup array with no `$sort`. If a condition address ever resolves to more than one installed plugin (reused condition, or a duplicate install not yet marked uninstalled), or a plugin has two `status=active` settings, the branch we pick — and the `token`/`minVotingPower` we return — is whatever Mongo hands back first, i.e. nondeterministic run-to-run.

Probably fine given how the indexer maintains these, but we could add a deterministic tiebreak (e.g. `$sort` by `blockNumber` desc in those lookup pipelines and take newest) so it's at least stable. Not blocking — more of a "let's not get surprised later" note.

File: src/models/schema/daoPermission.ts [hunk 5]
```diff
@@ -212,6 +218,118 @@ export default class DaoPermission extends Model {
...
+                    },
+                  ],
+                  default: { conditionType: 'unknown' },
+                },
+              },
+            },
+          },
+        },
+      },
+      { $addFields: { conditionAddress: { $ifNull: ['$conditionAddress', ALLOW_FLAG] } } },
```
Comment (thekidnamedkd) [inline src/models/schema/daoPermission.ts:342]: Just flagging for the PR description / consumers: `findWithPagination` used to return the true stored `conditionAddress` (`null` when unconditional); now `null` becomes the `0x…02` ALLOW_FLAG sentinel and `condition` is omitted. That's the intended new contract (and the FE for this feature relies on exactly that 👍) — but any existing consumer detecting unconditional grants via `conditionAddress == null` will now see a non-null value. Might be worth a line in the description so nobody's caught out.

File: src/types/routers.ts [hunk 6]
```diff
@@ -355,6 +355,27 @@ export interface IExecuteSelectorResponse {
...
+export interface IPermissionConditionData {
+  conditionType: string
+  token?: HexAddress
+  minVotingPower?: string
+  onlyListed?: boolean
+  selectors?: (string | null)[]
+  targets?: HexAddress[]
+}
+
+export interface IPermissionResponse {
```
Comment (thekidnamedkd) [inline src/types/routers.ts:368]: Are these wired up anywhere? `findWithPagination` is still `IPaginatedResult<any>` / returns `data as any`, and the controller doesn't reference `IPermissionResponse`, so the aggregation output isn't actually checked against these — a field rename in the pipeline (`condition` → `conditionData`) would compile clean and ship a broken shape. Could we type the controller return (or the mapped result) as `IPermissionResponse` so the interfaces earn their keep? Otherwise they'll quietly drift from what the pipeline emits.

**Review by thekidnamedkd — COMMENTED** [review]
Looks great — some massive work here! 🙌 Can we get what's already here up in sandbox ASAP so FE can start integrating? I left a few inline notes below, mostly edge cases around condition-type resolution and a couple of consistency nits — none are blockers for getting it deployed, we can follow up on them.

File: src/models/schema/daoPermission.ts [hunk 7]
```diff
@@ -212,6 +218,118 @@ export default class DaoPermission extends Model {
...
+                      },
+                    },
+                    {
+                      case: {
+                        $and: [
+                          { $eq: ['$$pp.matchedProposal', true] },
+                          { $eq: ['$$pp.interfaceType', IPluginInterfaceType.multisig] },
+                        ],
+                      },
+                      then: { conditionType: 'membership', onlyListed: '$$ps.onlyListed' },
```
Comment (thekidnamedkd) [inline src/models/schema/daoPermission.ts]: Could we also project `minApprovals` onto the membership condition? The design surfaces a threshold here ("Member of multisig: N"), and `onlyListed` alone can't produce that number — `minApprovals` is already on `Setting`, just not in the `$project` up at L275, so it's a cheap add:

```suggestion
                      then: { conditionType: 'membership', onlyListed: '$$ps.onlyListed', minApprovals: '$$ps.minApprovals' },
```
(plus `minApprovals: 1` in the L275 projection).

For the actual member *count*, is the intent that FE pulls it from the members endpoint rather than surfacing it here? It lives per-`pluginMember` rather than on `Setting`, so I'd rather not join it into this aggregation unless you think it belongs on the condition. Just want to make sure we're not double-sourcing it. 🙂

**Review by thekidnamedkd — CHANGES_REQUESTED** [review]
Requesting changes on a single item — everything else here is good to go and can ship to sandbox as-is.

**Blocker:** the membership condition only returns `onlyListed`, but the design needs a threshold — "Member of multisig: N" — which FE can't render from `onlyListed` alone. `minApprovals` is already on `Setting`, so it's a small additive change (details + `suggestion` in the inline thread on the membership branch). Flagging as Request Changes because this is a product-AC gap FE is holding on, not because the code is wrong.

Everything else is non-blocking: the correctness/consistency notes from the first pass are follow-up-able, and the no-condition + per-variant field contracts are already test-locked in `daoPermission.spec.ts`. Just need `minApprovals` (and a confirm on where member *count* should come from) before this satisfies the membership AC. Great work otherwise! 🙌

**Review by thekidnamedkd — APPROVED** [review]
thanks for the changes lgtm 🔥
